### DISCUSSION TITLE
Is there any extension or module for GOOD Prompt?

### DISCUSSION THREAD

**Discussion by oldhand7** [lllyasviel/Fooocus#3337] [Q&A]
I 've run Foocus with various prompt and found it's a prompt sensitive even though they are same meaning.
Sometimes it works well as thoughts but often far away from expectation.
So I tried to  use ChatGPT to generate GOOD prompt for Foocus then try -> result was good.

Could you please let me know how to pre-cook the prompts from the users so generate good output?

**Comment by mashb1t** [comment]
Please elaborate " pre-cook", do you mean model training? Using the style Fooocus V2 usually helps generating better results than without it.

**Comment by oldhand7** [comment]
Prompt1: A classroom personal space designed for reflection, analytical thinking, and student wellbeing. It features couches, booths, movable whiteboards, acoustic screens, lowered ceiling elements, window seats, and defined nooks.
![2024-07-22_13-14-02_5096](https://github.com/user-attachments/assets/2d5574cc-c7ce-40c4-988c-7c03d6ef1760)

Prompt2:  Personal Space fosters reflection, analytical thinking, and student wellbeing with furnishings like couches and booths, space dividers such as movable whiteboards and acoustic screens, lowered ceiling elements, window seats, and nooks defined by fixed or movable edges.
![Personal](https://github.com/user-attachments/assets/e3209956-eb16-4f63-84b8-25d89ed8eb02)



As you can see those two prompts are same but the results are completely differnet.


What I want to know is how to make the prompt well so that the AI don't misunderstand, not missing what I want to generate.

**Reply by mashb1t** [reply]
You can create or search for a LoRA for classrooms.
Your second prompt does not feature any subject, which is why the model does not generate anything resmembling a room.

As already hinted at multiple times before, please read https://stable-diffusion-art.com/prompt-guide/ and provide a subject in your prompts.

"Pre-bake" prompts => training, where you have to read into how to do this e.g. with kohja_ss or similar tools.

**Reply by oldhand7** [reply]
Thank you. I just want to know how to convert prompt into stable diffusion prompt. Is there some modules or engine for it?

**Reply by mashb1t** [reply]
No, you have to learn how to write a good prompt.
